### ISSUE TITLE
Just want to ask a question for education purposes... hope it's okay!

### ISSUE THREAD

**Issue by slerman12** [martinsbruveris/tensorflow-image-models#59]
Hi, this library is great. I'm wondering, maybe you can tell me this with your experience, what would be your small-list of the "top" essentials of a state-of-art image model these days? Data augmentation of course (mixup?), regularization (layer norm?), EMA, weight decay... I want to get a minimalist competitive ImageNet model working.

**Comment by slerman12** [comment]
Thanks. I'm building a library and was wondering if you had input

**Comment by slerman12** [comment]
Thanks. I'm building a library and was wondering if you had input.

**Comment by slerman12** [comment]
Please let me know if you think of any additions that would be absolute essentials - I have loosely read both of those papers, but I'll take a deeper look.

Implementation is hard, even when you know the recipe 😆

**Comment by martinsbruveris** [comment]
Implementation is always hard... Otherwise coding wouldn't be so much fun. For a certain notion of fun...

Good luck with your library! Is it open source?

**Comment by slerman12** [comment]
TBA, haha
